### PR TITLE
fix(tts): chunk long texts for Deepgram to bypass 2000-char limit

### PR DESCRIPTION
**Pull request by meteyou**
Deepgram's /v1/speak endpoint hard-rejects payloads larger than 2000 characters, causing the 'Read aloud' button to silently fail on any longer assistant message.

Add chunkTextForDeepgram() which splits at sentence/word boundaries (lookbehind on [.!?\n]) and falls back to word- then hard-splits for pathological inputs. synthesizeDeepgramWrapped() now iterates chunks, synthesizes each, and concatenates the resulting Buffers:

- mp3  → Buffer.concat() (MP3 frames are self-contained, concat is safe)
- wav  → chunks requested as linear16 (raw PCM), concatenated, then a single WAV header is prepended at the end
- opus / flac → explicit error with actionable message (container headers per-chunk make naive concat produce corrupt files)

The low-level synthesizeDeepgram() retains its strict 2000-char guard so existing tests are unaffected.

### CODE REVIEW

**Comment by prime-axiom** [comment]
Running a full checklist review now (including local checks). I’ll post a structured review shortly.

File: packages/core/src/tts.ts [hunk 1]
```diff
@@ -323,10 +379,29 @@ async function synthesizeDeepgramWrapped(
...
+  const model = options.voice || settings.deepgramModel
+
+  // Deepgram caps `/v1/speak` at 2000 chars per call. Split longer texts
+  // and concatenate the audio. Safe for `mp3` (frame-aligned) and
+  // `linear16` (raw PCM — just bytes); `opus`/`flac` use page/frame
+  // containers that don't survive naive concatenation, so we refuse those
+  // explicitly with an actionable error rather than producing a corrupt
+  // file.
+  const chunks = chunkTextForDeepgram(text)
+  if (chunks.length > 1 && (deepgramEncoding === 'opus' || deepgramEncoding === 'flac')) {
```
Comment (prime-axiom) [inline packages/core/src/tts.ts]: This rejects valid `opus`/`flac` inputs between 1901 and 2000 chars because `chunkTextForDeepgram()` splits at 1900, while the error says `>2000` and Deepgram's limit is 2000. Gate this on the actual trimmed text length instead of `chunks.length`.

```suggestion
  if (text.trim().length > 2000 && (deepgramEncoding === 'opus' || deepgramEncoding === 'flac')) {
```

File: packages/core/src/tts.ts [hunk 2]
```diff
@@ -40,6 +40,62 @@ const DEEPGRAM_LINEAR16_SAMPLE_RATE = 24000
...
+const DEEPGRAM_TTS_CHUNK_LIMIT = 1900
+
+/**
+ * Split `text` into chunks of at most `maxLen` characters, preferring
+ * paragraph → sentence → word boundaries before falling back to a hard
+ * slice. The goal is to keep prosody intact across chunks: cutting at
+ * `. ` / `! ` / `? ` / `\n\n` produces audibly smoother joins than
+ * mid-sentence splits.
+ */
+export function chunkTextForDeepgram(text: string, maxLen = DEEPGRAM_TTS_CHUNK_LIMIT): string[] {
```
Comment (prime-axiom) [inline packages/core/src/tts.ts]: This export appears to exist only for future/direct tests. The repository enforces Fallow dead-code checks, so exported helpers need either a production consumer or a test that actually imports them. If you do not add direct unit tests for this helper in this PR, keep it internal.

```suggestion
function chunkTextForDeepgram(text: string, maxLen = DEEPGRAM_TTS_CHUNK_LIMIT): string[] {
```

**Review by prime-axiom — CHANGES_REQUESTED** [review]
I found a few merge-blocking issues. The implementation direction is good, but this needs one threshold fix, the CI/dead-code issue addressed, and docs/tests for the new user-visible behavior before it is merge-ready.

Blockers:
- Deepgram `opus`/`flac` requests between 1901 and 2000 characters are rejected even though Deepgram should accept up to 2000 chars, because the branch checks `chunks.length > 1` after chunking at 1900.
- `chunkTextForDeepgram` is exported only for tests but currently unused by production imports, which trips the repository's dead-code gate.
- The PR changes user-visible TTS behavior and adds an inline chat error banner, but does not update the user-facing docs. Please document the long-text behavior and the `mp3`/`wav` limitation for Deepgram chunking.

Also please add regression tests for the chunking boundary and unsupported-format behavior so this does not regress later.

**Review by prime-axiom — APPROVED** [review]
Checklist passed on d043374: the Deepgram boundary fixes, concat-safe mp3/wav chunking, opus/flac rejection path, regression tests, docs, and green CI all look merge-ready.
